lib/rubocop/cop/rspec/factory_bot/static_attribute_defined_dynamically.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      module FactoryBot
+        # Prefer declaring static attribute values without a block.
+        #
+        # @see DynamicAttributeDefinedStatically
+        #
+        # @example
+        #   # bad
+        #   kind { :static }
+        #
+        #   # good
+        #   kind :static
+        #
+        #   # bad
+        #   comments_count { 0 }
+        #
+        #   # good
+        #   comments_count 0
+        #
+        #   # bad
+        #   type { User::MAGIC }
+        #
+        #   # good
+        #   type User::MAGIC
+        class StaticAttributeDefinedDynamically < Cop
+          MSG = 'Dont use a block to set a static value to an attribute.'.freeze
+
+          def_node_matcher :block_value_matcher, <<-PATTERN
+          (block (send nil? _) _ $...)
+          PATTERN
+
+          def_node_search :factory_attributes, <<-PATTERN
+          (block (send nil? { :factory :trait } ...) _ { (begin $...) $(send ...) $(block ...) } )
+          PATTERN
+
+          def on_block(node)
+            factory_attributes(node).to_a.flatten.each do |attribute|
+              values = block_value_matcher(attribute)
+              next if values.nil? ||
```

Comment:
encapsulate this in a method. Are you sure you should check for `none?` here

Reply:
Yes, we want to add an offence if there are any 'static' nodes matched (ie next if none).

Will change to `next if values.to_a.none? { |v| static?(v) }` and encapsulate the static checks into a method